config/route53/config.go

```diff
@@ -0,0 +1,124 @@
+
+import (
+	"strings"
+
+	"github.com/crossplane-contrib/terrajet/pkg/config"
+)
+
+// Configure route53 resources.
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("aws_route53_delegation_set", func(r *config.Resource) {
+		r.ExternalName = config.IdentifierFromProvider
+	})
+	p.AddResourceConfigurator("aws_route53_health_check", func(r *config.Resource) {
+		r.ExternalName = config.IdentifierFromProvider
+	})
+	p.AddResourceConfigurator("aws_route53_hosted_zone_dnssec", func(r *config.Resource) {
+		r.ExternalName = config.IdentifierFromProvider
+		r.References = config.References{
+			"hosted_zone_id": config.Reference{
+				Type: "Zone",
+			},
+		}
+	})
+	p.AddResourceConfigurator("aws_route53_key_signing_key", func(r *config.Resource) {
+		r.ExternalName = config.IdentifierFromProvider
+		r.References = config.References{
+			"hosted_zone_id": config.Reference{
+				Type: "Zone",
+			},
+			"key_management_service_arn": config.Reference{
+				Type: "github.com/crossplane-contrib/provider-jet-aws/apis/kms/v1alpha1.Key",
+			},
+		}
+	})
+	p.AddResourceConfigurator("aws_route53_query_log", func(r *config.Resource) {
+		r.ExternalName = config.IdentifierFromProvider
+		r.References = config.References{
+			"hosted_zone_id": config.Reference{
+				Type: "Zone",
+			},
+		}
+	})
+	p.AddResourceConfigurator("aws_route53_record", func(r *config.Resource) {
+		r.ExternalName = config.IdentifierFromProvider
+		r.References = config.References{
+			"zone_id": config.Reference{
+				Type: "Zone",
+			},
+			"health_check_id": config.Reference{
+				Type: "HealthCheck",
+			},
+		}
+	})
+	p.AddResourceConfigurator("aws_route53_vpc_association_authorization", func(r *config.Resource) {
+		r.ExternalName = config.IdentifierFromProvider
+		r.ExternalName.SetIdentifierArgumentFn = func(base map[string]interface{}, externalName string) {
+			words := strings.Split(externalName, ":")
+			if len(words) != 2 {
+				return
+			}
+			base["zone_id"] = words[0]
+			base["vpc_id"] = words[1]
+		}
+		r.References = config.References{
+			"zone_id": config.Reference{
+				Type: "Zone",
+			},
+			"vpc_id": config.Reference{
+				Type: "github.com/crossplane-contrib/provider-jet-aws/apis/ec2/v1alpha1.VPC",
+			},
+		}
+	})
+	p.AddResourceConfigurator("aws_route53_zone", func(r *config.Resource) {
+		r.ExternalName = config.IdentifierFromProvider
+		r.References = config.References{
+			"delegation_set_id": config.Reference{
+				Type: "DelegationSet",
+			},
+			"vpc.vpc_id": config.Reference{
+				Type: "github.com/crossplane-contrib/provider-jet-aws/apis/ec2/v1alpha1.VPC",
+			},
+		}
+	})
+	p.AddResourceConfigurator("aws_route53_zone_association", func(r *config.Resource) {
+		r.ExternalName = config.IdentifierFromProvider
+		// Z123456ABCDEFG:vpc-12345678
+		// Z123456ABCDEFG:vpc-12345678:us-east-2
+		r.ExternalName.SetIdentifierArgumentFn = func(base map[string]interface{}, externalName string) {
+			words := strings.Split(externalName, ":")
```

Comment:
Like we discussed for other configuration functions, being able to report errors really paves off. But this is not an issue of this PR.

Reply:
It's hard to know whether it's an error case though. For example, in this case the ID comes from the provider, hence it can be empty until creation finishes so until then there is no data to set and it's a valid flow. So, specifically for this `SetIdentifierArgumentFn`, having a best-effort approach is more inline than returning error IMO.